Mapper::insertFuture has compile error

**Describe the bug**
When I was using the orm, and use the insertFuture. It seemed that it can not be compiled pass.

**To Reproduce**
Steps to reproduce the behavior:
1. I just use Mapper<T>::insertFuture, and there are some complie error.

**Expected behavior**
just can be compiled correctly

**Screenshots**
here are the compile error, it seemed that the exception production function's conversion was failed.
```
[ 16%] Building CXX object CMakeFiles/TGUCSDN1.dir/controllers/UserManager.cc.o
In file included from /root/TGUCSDN1/controllers/../models/User.h:13:0,
                 from /root/TGUCSDN1/controllers/UserManager.cc:2:
/usr/local/include/drogon/orm/Mapper.h: In instantiation of 'drogon::orm::Mapper<T>::insertFuture(const T&)::<lambda(const drogon::orm::Result&)> [with T = drogon_model::tgucsdn1::User]':
/usr/local/include/drogon/orm/Mapper.h:1221:35:   required from 'struct drogon::orm::Mapper<T>::insertFuture(const T&) [with T = drogon_model::tgucsdn1::User]::<lambda(const class drogon::orm::Result&)>'
/usr/local/include/drogon/orm/Mapper.h:1221:12:   required from 'std::future<_Res> drogon::orm::Mapper<T>::insertFuture(const T&) [with T = drogon_model::tgucsdn1::User]'
/root/TGUCSDN1/controllers/UserManager.cc:80:29:   required from here
/usr/local/include/drogon/orm/Mapper.h:1243:17: error: no matching function for call to 'drogon::orm::Mapper<drogon_model::tgucsdn1::User>::findByPrimaryKey(const PrimaryKeyType&, drogon::orm::Mapper<T>::insertFuture(const T&)::<lambda(const drogon::orm::Result&)> [with T = drogon_model::tgucsdn1::User]::<lambda(drogon_model::tgucsdn1::User)>, drogon::orm::Mapper<T>::insertFuture(const T&)::<lambda(const drogon::orm::Result&)> [with T = drogon_model::tgucsdn1::User]::<lambda(const std::__exception_ptr::exception_ptr&)>)'
                 tmp.findByPrimaryKey(
                 ^~~
/usr/local/include/drogon/orm/Mapper.h:187:5: note: candidate: template<class U> typename std::enable_if<(! std::is_same<typename U::PrimaryKeyType, void>::value), T>::type drogon::orm::Mapper<T>::findByPrimaryKey(const TraitsPKType&) [with U = U; T = drogon_model::tgucsdn1::User]
     findByPrimaryKey(const TraitsPKType &key) noexcept(false)
     ^~~~~~~~~~~~~~~~
/usr/local/include/drogon/orm/Mapper.h:187:5: note:   template argument deduction/substitution failed:
/usr/local/include/drogon/orm/Mapper.h:1243:17: note:   candidate expects 1 argument, 3 provided
                 tmp.findByPrimaryKey(
                 ^~~
/usr/local/include/drogon/orm/Mapper.h:227:5: note: candidate: template<class U> typename std::enable_if<std::is_same<typename U::PrimaryKeyType, void>::value, T>::type drogon::orm::Mapper<T>::findByPrimaryKey(const TraitsPKType&) [with U = U; T = drogon_model::tgucsdn1::User]
     findByPrimaryKey(const TraitsPKType &key) noexcept(false)
     ^~~~~~~~~~~~~~~~
/usr/local/include/drogon/orm/Mapper.h:227:5: note:   template argument deduction/substitution failed:
/usr/local/include/drogon/orm/Mapper.h:1243:17: note:   candidate expects 1 argument, 3 provided
                 tmp.findByPrimaryKey(
                 ^~~
/usr/local/include/drogon/orm/Mapper.h:244:5: note: candidate: template<class U> typename std::enable_if<(! std::is_same<typename U::PrimaryKeyType, void>::value), void>::type drogon::orm::Mapper<T>::findByPrimaryKey(const TraitsPKType&, const SingleRowCallback&, const ExceptionCallback&) [with U = U; T = drogon_model::tgucsdn1::User]
     findByPrimaryKey(const TraitsPKType &key,
     ^~~~~~~~~~~~~~~~
/usr/local/include/drogon/orm/Mapper.h:244:5: note:   template argument deduction/substitution failed:
/usr/local/include/drogon/orm/Mapper.h:1243:17: note:   cannot convert '<lambda closure object>drogon::orm::Mapper<T>::insertFuture(const T&)::<lambda(const drogon::orm::Result&)> [with T = drogon_model::tgucsdn1::User]::<lambda(const std::__exception_ptr::exception_ptr&)>{std::shared_ptr<std::promise<drogon_model::tgucsdn1::User> >(prom)}' (type 'drogon::orm::Mapper<T>::insertFuture(const T&)::<lambda(const drogon::orm::Result&)> [with T = drogon_model::tgucsdn1::User]::<lambda(const std::__exception_ptr::exception_ptr&)>') to type 'const ExceptionCallback& {aka const std::function<void(const drogon::orm::DrogonDbException&)>&}'
                 tmp.findByPrimaryKey(
                 ^~~
/usr/local/include/drogon/orm/Mapper.h:284:5: note: candidate: template<class U> typename std::enable_if<std::is_same<typename U::PrimaryKeyType, void>::value, void>::type drogon::orm::Mapper<T>::findByPrimaryKey(const TraitsPKType&, const SingleRowCallback&, const ExceptionCallback&) [with U = U; T = drogon_model::tgucsdn1::User]
     findByPrimaryKey(const TraitsPKType &key,
     ^~~~~~~~~~~~~~~~
/usr/local/include/drogon/orm/Mapper.h:284:5: note:   template argument deduction/substitution failed:
/usr/local/include/drogon/orm/Mapper.h:1243:17: note:   cannot convert '<lambda closure object>drogon::orm::Mapper<T>::insertFuture(const T&)::<lambda(const drogon::orm::Result&)> [with T = drogon_model::tgucsdn1::User]::<lambda(const std::__exception_ptr::exception_ptr&)>{std::shared_ptr<std::promise<drogon_model::tgucsdn1::User> >(prom)}' (type 'drogon::orm::Mapper<T>::insertFuture(const T&)::<lambda(const drogon::orm::Result&)> [with T = drogon_model::tgucsdn1::User]::<lambda(const std::__exception_ptr::exception_ptr&)>') to type 'const ExceptionCallback& {aka const std::function<void(const drogon::orm::DrogonDbException&)>&}'
                 tmp.findByPrimaryKey(
                 ^~~
In file included from /usr/local/include/drogon/orm/DbClient.h:22:0,
                 from /usr/local/include/drogon/HttpAppFramework.h:30,
                 from /usr/local/include/drogon/HttpController.h:19,
                 from /root/TGUCSDN1/controllers/UserManager.h:2,
                 from /root/TGUCSDN1/controllers/UserManager.cc:1:
/usr/local/include/drogon/orm/SqlBinder.h:317:65: error: 'typename std::enable_if<traits:: isSqlCallback, drogon::orm::internal::SqlBinder>::type& drogon::orm::internal::SqlBinder::operator>>(CallbackType&&) [with CallbackType = drogon::orm::Mapper<T>::insertFuture(const T&) [with T = drogon_model::tgucsdn1::User]::<lambda(const drogon::orm::Result&)>; traits = drogon::orm::internal::FunctionTraits<drogon::orm::Mapper<T>::insertFuture(const T&) [with T = drogon_model::tgucsdn1::User]::<lambda(const drogon::orm::Result&)> >; typename std::enable_if<traits:: isSqlCallback, drogon::orm::internal::SqlBinder>::type = drogon::orm::internal::SqlBinder]', declared using local type 'drogon::orm::Mapper<T>::insertFuture(const T&) [with T = drogon_model::tgucsdn1::User]::<lambda(const drogon::orm::Result&)>', is used but never defined [-fpermissive]
     typename std::enable_if<traits::isSqlCallback, self>::type &operator>>(
                                                                 ^~~~~~~~
/usr/local/include/drogon/orm/SqlBinder.h:296:5: error: 'typename std::enable_if<(traits:: isExceptCallback && traits:: isPtr), drogon::orm::internal::SqlBinder>::type& drogon::orm::internal::SqlBinder::operator>>(CallbackType&&) [with CallbackType = drogon::orm::Mapper<T>::insertFuture(const T&) [with T = drogon_model::tgucsdn1::User]::<lambda(const std::__exception_ptr::exception_ptr&)>; traits = drogon::orm::internal::FunctionTraits<drogon::orm::Mapper<T>::insertFuture(const T&) [with T = drogon_model::tgucsdn1::User]::<lambda(const std::__exception_ptr::exception_ptr&)> >; typename std::enable_if<(traits:: isExceptCallback && traits:: isPtr), drogon::orm::internal::SqlBinder>::type = drogon::orm::internal::SqlBinder]', declared using local type 'drogon::orm::Mapper<T>::insertFuture(const T&) [with T = drogon_model::tgucsdn1::User]::<lambda(const std::__exception_ptr::exception_ptr&)>', is used but never defined [-fpermissive]
     operator>>(CallbackType &&callback)
     ^~~~~~~~
CMakeFiles/TGUCSDN1.dir/build.make:110: recipe for target 'CMakeFiles/TGUCSDN1.dir/controllers/UserManager.cc.o' failed
make[2]: *** [CMakeFiles/TGUCSDN1.dir/controllers/UserManager.cc.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/TGUCSDN1.dir/all' failed
make[1]: *** [CMakeFiles/TGUCSDN1.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```
**Desktop (please complete the following information):**
 - OS: Ubuntu

--- a/orm_lib/inc/drogon/orm/Mapper.h
+++ b/orm_lib/inc/drogon/orm/Mapper.h
@@ -1243,8 +1243,15 @@ inline std::future<T> Mapper<T>::insertFuture(const T &obj) noexcept
                 tmp.findByPrimaryKey(
                     newObj.getPrimaryKey(),
                     [prom](T selObj) { prom->set_value(selObj); },
-                    [prom](const std::exception_ptr &e) {
-                        prom->set_exception(e);
+                    [prom](const DrogonDbException &e) {
+                        try
+                        {
+                            throw e;
+                        }
+                        catch (...)
+                        {
+                            prom->set_exception(std::current_exception());
+                        }
                     });
             }
             else

--- a/orm_lib/tests/db_test.cc
+++ b/orm_lib/tests/db_test.cc
@@ -31,7 +31,7 @@ using namespace drogon::orm;
 #define RED "\033[31m"   /* Red */
 #define GREEN "\033[32m" /* Green */
 
-constexpr int postgre_tests = 43;
+constexpr int postgre_tests = 44;
 constexpr int mysql_tests = 45;
 constexpr int sqlite_tests = 47;
 
@@ -650,13 +650,30 @@ void doPostgreTest(const drogon::orm::DbClientPtr &clientPtr)
             testOutput(false,
                        "postgresql - ORM mapper asynchronous interface(0)");
         });
+
+    /// 6.1.5 insert future
+    user.setUserId("pg_future");
+    auto fu = mapper.insertFuture(user);
+    try
+    {
+        auto u = fu.get();
+        testOutput(true,
+                   "postgresql - ORM mapper asynchronous future interface(0)");
+    }
+    catch (const DrogonDbException &e)
+    {
+        std::cerr << e.base().what() << std::endl;
+        testOutput(false,
+                   "postgresql - ORM mapper asynchronous future interface(0)");
+    }
+
     /// 6.2 insert
     user.setUserId("pg1");
     user.setUserName("postgres1");
     mapper.insert(
         user,
         [](Users ret) {
-            testOutput(ret.getPrimaryKey() == 2,
+            testOutput(ret.getPrimaryKey() == 3,
                        "postgresql - ORM mapper asynchronous interface(1)");
         },
         [](const DrogonDbException &e) {

